Document how to use mock in JVM mode tests #1449 (#1452)
apache/camel-quarkus: docs/modules/ROOT/pages/extensions/mock.adoc

----

Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.

== Usage

> To use camel-mock capabilities it is required to get access to MockEndpoint instances.

Reviewer: ```suggestion
To use camel-mock capabilities in tests it is required to get access to MockEndpoint instances.
```

CDI injection could be used for accessing instances (see https://quarkus.io/guides/getting-started-testing#injection-into-tests[Quarkus documentation]).
You can inject camelContext into test using `@Inject` annotation. Camel context can be then used for obtaining mock endpoints.
See the following example:

----

> @Inject

Reviewer: ```suggestion

@QuarkusTest
class MyTest {

@Inject
```

plus I vote for naming all or at least the most important imports.

CamelContext camelContext;

@Test
public void test() throws InterruptedException {

    //test code

    MockEndpoint mockEndpoint = camelContext.getEndpoint("mock:result", MockEndpoint.class);

>     mockEndpoint.expectedBodiesReceived("Hello World");

Reviewer: There should be some route invocation at this location (through rest or directly over Java API). Otherwise the test does not make much sense. It would also be nice to have the route definition in the example.

    mockEndpoint.assertIsSatisfied();
}
----

== Camel Quarkus limitations

Injection of CDI beans (described in Usage) does not work in native mode.

> In the native mode the test and the applications are running in two different processes and it is not possible

Reviewer: ```suggestion
In the native mode the test and the application under test are running in two different processes and it is not possible
```
